tests/test_amqp.py

```diff
@@ -1587,6 +1587,141 @@ async def test_heartbeat_disabling(
         async with connection:
             assert heartbeat == 0
 
+    async def test_non_acked_messages_are_redelivered_to_queue(
+        self,
+        channel: aio_pika.Channel,
+        declare_queue: Callable,
+        declare_exchange: Callable,
+    ):
+        queue_name = get_random_name("test_connection")
+        routing_key = get_random_name()
+
+        exchange = await declare_exchange(
+            "direct", auto_delete=True, channel=channel,
+        )
+
+        queue: AbstractQueue = await declare_queue(
+            queue_name, auto_delete=False, channel=channel,
+        )
+
+        await queue.bind(exchange, routing_key)
+
+        # Publish 5 messages to queue
+        all_bodies = []
+        for _ in range(0, 5):
+            body = bytes(shortuuid.uuid(), "utf-8")
+            all_bodies.append(body)
+
+            assert await exchange.publish(Message(body), routing_key)
+
+        # Create a subscription but only process first message
+        async with queue.iterator() as queue_iterator:
+            first_message = await anext(queue_iterator)
+            async with first_message.process():
+                assert first_message.body == all_bodies[0]
+
+        # Confirm other messages are still in queue
+        for i in range(1, 5):
+            incoming_message = await queue.get(timeout=5)
+            await incoming_message.ack()
+
+            assert incoming_message.body == all_bodies[i]
+
+        # Check if the queue is now empty
+        assert await queue.get(fail=False, timeout=.5) is None
+
+        # Cleanup, delete the queue
+        await queue.delete()
+
+    async def test_regression_only_messages_cancelled_subscription_are_nacked(
```

Comment:
Regression test which failed before fixing the code. Documentation is inline to show why the test is constructed the way it is.